libs/app-shell/src/lib/components/plugin-launcher/plugin-launcher.component.ts

```diff
@@ -98,31 +101,30 @@ export class PluginLauncherComponent implements OnChanges, AfterViewChecked {
             return;
         }
 
-        const _module = descriptor.modules.find(module => module.name === this.module);
-        const _component = await loadRemoteModule<AngularIvyComponentDescriptor>(descriptor, _module as AngularIvyComponentDescriptor)
-            .then(m => m[_module.name])
+        const _pluginModule = descriptor.modules.find(module => module.name === this.module);
+        const _remoteModule = await loadRemoteModule<DescriptorsModule>(descriptor, _pluginModule)
             .catch(err => console.error(err));
+        const _module = _remoteModule[_pluginModule.name];
 
-        if (_module.type !== 'iframe' && !this.iframeUri) {
+        if (_pluginModule.type !== 'iframe' && !this.iframeUri) {
             this._safeIframeUri = null;
         }
 
-        if (_module.type === 'iframe') {
-            const _url = descriptor.uri + _module.html;
+        if (_pluginModule.type === 'iframe') {
+            const _url = descriptor.uri + _pluginModule.html;
             this._safeIframeUri = this.sanitizer.bypassSecurityTrustResourceUrl(_url);
             return;
         }
 
-        if (_module.type === 'custom-element') {
-            const element = document.createElement(_component);
+        if (_pluginModule.type === 'custom-element') {
+            const element = document.createElement(_module);
             this._render.appendChild(this._elementRef.nativeElement, element);
             return;
         }
 
-        if (_module.type === 'angular-ivy-component' && this.ngContentView) {
-            this.ngContentView.clear();
-            const factory = this.cfr.resolveComponentFactory(_component);
-            this.ngContentView.createComponent(factory, null, this._injector);
+        if (_pluginModule.type === 'angular-ivy-component') {
```

Comment:
Does this changes also completly functionality that we discuss in the other thread? 
That we dont support exposing components anymore and we go fully to Modules only?

Reply:
it may work without moduleFactory, just with component